### PR TITLE
Fix state update issue in SMS Templates UI

### PR DESCRIPTION
**Pull request by Avarjana**
### Description
Fix state update issue for danger zone once a API call is happened.

### Related issue
- https://github.com/wso2/product-is/issues/21363

### CODE REVIEW

File: features/admin.sms-templates.v1/pages/sms-customization.tsx [hunk 1]
```diff
@@ -99,7 +99,12 @@ const SMSCustomizationPage: FunctionComponent<SMSCustomizationPageInterface> = (
         error: smsTemplatesListError
     } = useGetSmsTemplatesList();
 
-    const { data: smsTemplate, isLoading: isSmsTemplateLoading, error: smsTemplateError } = useGetSmsTemplate(
+    const {
+        data: smsTemplate,
+        isLoading: isSmsTemplateLoading,
+        error: smsTemplateError,
+        mutate: mutate
```
Comment (pavinduLakshan) [inline features/admin.sms-templates.v1/pages/sms-customization.tsx:106]:
```suggestion
        mutate: mutateSmsTemplates
```

File: features/admin.sms-templates.v1/pages/sms-customization.tsx [hunk 2]
```diff
@@ -208,6 +213,7 @@ const SMSCustomizationPage: FunctionComponent<SMSCustomizationPageInterface> = (
             availableSmsTemplatesList?.find((template: SMSTemplateType) => template.id === templateId)?.description
         );
         setShouldFetch(true);
+        mutate();
```
Comment (pavinduLakshan) [inline features/admin.sms-templates.v1/pages/sms-customization.tsx:216]:
```suggestion
        mutateSmsTemplates();
```

File: features/admin.sms-templates.v1/pages/sms-customization.tsx [hunk 3]
```diff
@@ -224,6 +230,7 @@ const SMSCustomizationPage: FunctionComponent<SMSCustomizationPageInterface> = (
         setIsSystemTemplate(false);
         setSelectedLocale(locale);
         setShouldFetch(true);
+        mutate();
```
Comment (pavinduLakshan) [inline features/admin.sms-templates.v1/pages/sms-customization.tsx:233]:
```suggestion
        mutateSmsTemplates();
```

File: features/admin.sms-templates.v1/pages/sms-customization.tsx [hunk 4]
```diff
@@ -245,6 +252,7 @@ const SMSCustomizationPage: FunctionComponent<SMSCustomizationPageInterface> = (
                     );
                     setIsSystemTemplate(false);
                     setShouldFetch(true);
+                    mutate();
```
Comment (pavinduLakshan) [inline features/admin.sms-templates.v1/pages/sms-customization.tsx:255]:
```suggestion
                    mutateSmsTemplates();
```

File: features/admin.sms-templates.v1/pages/sms-customization.tsx [hunk 5]
```diff
@@ -267,6 +275,7 @@ const SMSCustomizationPage: FunctionComponent<SMSCustomizationPageInterface> = (
...
```
Comment (pavinduLakshan) [inline features/admin.sms-templates.v1/pages/sms-customization.tsx:278]:
```suggestion
                    mutateSmsTemplates();
```

File: features/admin.sms-templates.v1/pages/sms-customization.tsx [hunk 6]
```diff
@@ -297,6 +306,7 @@ const SMSCustomizationPage: FunctionComponent<SMSCustomizationPageInterface> = (
                 setSelectedLocale(SMSTemplateConstants.DEAFULT_LOCALE);
                 setIsSystemTemplate(false);
                 setShouldFetch(true);
+                mutate();
```
Comment (pavinduLakshan) [inline features/admin.sms-templates.v1/pages/sms-customization.tsx:309]:
```suggestion
                mutateSmsTemplates();
```

**Review by pavinduLakshan — COMMENTED** [review]
Let's use a context-aware name for the `mutate` function. This is because multiple mutate calls from different hooks might exist in the same file, and future readers shouldn't have to trace back to the definition to determine which API each mutate call relates to.

File: features/admin.sms-templates.v1/pages/sms-customization.tsx (hunk 6 above)
Comment (Avarjana) [reply features/admin.sms-templates.v1/pages/sms-customization.tsx]: https://github.com/wso2/identity-apps/pull/7054/commits/aa3156ecb85b3bc6efd9318a37f37983ba801824

File: features/admin.sms-templates.v1/pages/sms-customization.tsx (hunk 5 above)
Comment (Avarjana) [reply features/admin.sms-templates.v1/pages/sms-customization.tsx]: https://github.com/wso2/identity-apps/pull/7054/commits/aa3156ecb85b3bc6efd9318a37f37983ba801824

File: features/admin.sms-templates.v1/pages/sms-customization.tsx (hunk 4 above)
Comment (Avarjana) [reply features/admin.sms-templates.v1/pages/sms-customization.tsx]: https://github.com/wso2/identity-apps/pull/7054/commits/aa3156ecb85b3bc6efd9318a37f37983ba801824

File: features/admin.sms-templates.v1/pages/sms-customization.tsx (hunk 2 above)
Comment (Avarjana) [reply features/admin.sms-templates.v1/pages/sms-customization.tsx]: https://github.com/wso2/identity-apps/pull/7054/commits/aa3156ecb85b3bc6efd9318a37f37983ba801824

File: features/admin.sms-templates.v1/pages/sms-customization.tsx (hunk 1 above)
Comment (Avarjana) [reply features/admin.sms-templates.v1/pages/sms-customization.tsx]: https://github.com/wso2/identity-apps/pull/7054/commits/aa3156ecb85b3bc6efd9318a37f37983ba801824

**Review by pavinduLakshan — APPROVED** [review]

**Pull request merged**